packages/rancher-alerting-drivers/charts/templates/hardened.yaml

```diff
@@ -0,0 +1,117 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ include "drivers.fullname" . }}-patch-sa
+  namespace: {{ .Release.Namespace }}
+  labels: {{ include "drivers.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": post-install, post-upgrade
+    "helm.sh/hook-delete-policy": hook-succeeded, before-hook-creation
+spec:
+  backoffLimit: 1
+  template:
+    spec:
+      serviceAccountName: {{ include "drivers.fullname" . }}-patch-sa
+      securityContext:
+        runAsNonRoot: true
+        runAsUser: 1000
+      restartPolicy: Never
+      nodeSelector: {{ include "linux-node-selector" . | nindent 8 }}
+      tolerations: {{ include "linux-node-tolerations" . | nindent 8 }}
+      containers:
+        - name: {{ include "drivers.fullname" . }}-patch-sa
+          image: "{{ include "system_default_registry" . }}{{ .Values.global.kubectl.repository }}:{{ .Values.global.kubectl.tag }}"
+          imagePullPolicy: IfNotPresent
+          command: ["kubectl", "patch", "serviceaccount", "default", "-p", "{\"automountServiceAccountToken\": false}"]
```

Comment:
Is this the only way to make hardened cluster pass? By changing the default service account ? I haven’t seen this pattern before. cc @aiyengar2

Reply:
@cbron you need to patch the default service account in any namespace that you create as part of the chart to be un-automounted. Declared as a req in https://rancher.com/docs/rancher/v2.x/en/security/rancher-2.5/1.6-hardening-2.5/#set-automountserviceaccounttoken-to-false-for-default-service-accounts.

Reply:
Why is the namespace in the arg instead of the command ? Also is specifying the namespace required if the SA is from the same namespace ? I'm not sure on that one, depends how default SA and kubectl act.

Reply:
The job template is from https://github.com/rancher/charts/blob/dev-v2.5-source/packages/rancher-cis-benchmark/charts/templates/patch_default_serviceaccount.yaml

I agree that it is not necessary to pass the ns as an arg. I will update the PR.

Reply:
ok lets go with passing explicitly in the command

Reply:
Fixed: the arg values were moved to the command.